src/Lachain.Core/RPC/HTTP/Web3/TransactionServiceWeb3.cs

```diff
@@ -540,10 +544,33 @@ public string Call(JObject opts, string? blockId)
                 : throw new Exception("Error in contract call");
         }
 
+        private bool IsGreater(ulong? lastBlockGasUsed, ulong percent, ulong gasLimit)
+        {
+            return lastBlockGasUsed * 100 > percent * gasLimit;
+        }
+
         [JsonRpcMethod("eth_gasPrice")]
         public string GetNetworkGasPrice()
         {
-            return Web3DataFormatUtils.Web3Number(_stateManager.CurrentSnapshot.NetworkGasPrice.ToUInt256());
+            var gasLimit = GasMetering.DefaultBlockGasLimit;
+
+            var latestBlock = _blockManager.LatestBlock();
+            var previousBlockHash = latestBlock.Hash.ToHex();
+            var lastBlockGasUsed = GetGasUsed(previousBlockHash);
```

Comment:
We should use last block here,  not its parent

Reply:
Sure, my bad fixing it :(

Reply:
So,  how about the fix?